mail_activity_reminder/models/mail_activity.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from pytz import timezone, UTC
+
+from odoo import _, api, fields, models
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    last_reminder = fields.Date()
+
+    @api.model
+    def _get_activities_to_remind(self):
+        utc_now = self.env.context.get(
+            'mail_activity_reminder_now',
+            fields.Datetime.now()
+        ).replace(tzinfo=UTC)
+
+        activities_to_remind = self.env['mail.activity']
+        for activity in self.search([]):
+            tz = activity.user_id.sudo().tz or 'UTC'
+
+            local_today = utc_now.astimezone(timezone(tz)).date()
+            local_deadline = activity.date_deadline
+            if local_deadline < local_today:
+                continue
+
+            reminders = activity.activity_type_id._get_reminder_offsets()
+            if not reminders:
+                continue
+
+            reminder = (local_deadline - local_today).days
+            if reminder not in reminders:
+                continue
+
+            if activity.last_reminder \
+                    and activity.last_reminder >= local_today:
+                continue
+
+            activities_to_remind |= activity
+        return activities_to_remind
+
+    @api.model
+    def _process_reminders(self):
+        activities = self._get_activities_to_remind()
+        activities.action_remind()
+        return activities
+
+    @api.multi
+    def action_notify(self):
+        super().action_notify()
+
+        utc_now = self.env.context.get(
+            'mail_activity_reminder_now',
+            fields.Datetime.now()
+        ).replace(tzinfo=UTC)
+
+        for activity in self.filtered(lambda x: not x.last_reminder):
+            tz = activity.user_id.sudo().tz or 'UTC'
+            local_today = utc_now.astimezone(timezone(tz)).date()
+            activity.last_reminder = local_today
+
+    @api.multi
+    def action_remind(self):
+        IrModel = self.env['ir.model']
+        MailThread = self.env['mail.thread']
+        message_activity_assigned = self.env.ref(
+            'mail.message_activity_assigned'
+        )
+        utc_now = self.env.context.get(
```

Comment:
Used 3 times already: worth to move this to `_reminder_utc_now` method?

Reply:
that feels a bit too much, as it does not have any specific logic except for giving ability to override value for testing

Reply:
then see my comment on tests ;)